### ISSUE TITLE
Need to reauthorize every day?

### ISSUE THREAD

**Issue by GsynthChris** [tuxuser/hassio-addons#21]
Hello, I've been having some trouble with this integration for the last few weeks. I found a sort of work around but it seems to just be a daily bandaid.

Once a day when I try to control my xbox via rest commands I have to do the following to "reconnect" my xbox. I go to `http://192.168.29.5:5557/device/FD00CFA257157C78/connect`
the first time the link loads I see `{"message":"Message '5b760e83.5': 'Message.CONFIGURATION' got no response!","success":false}` {"message":"Message '5b760e83.5': 'Message.CONFIGURATION' got no response!","success":false}and then I refresh the page and on the second try I get `{"connection_state":"Connected","success":true}`. 

It almost always works on the second try, but when it doesn't I go to `http://192.168.29.5:5557/auth/oauth` and I have to reauthorize my account. Each day I have to go to the connect link twice, about 1-2 times a week I have to go to the auth/oauth link and reconnect my account.

It seems to only happen after the xbox hasn't been used for a while. Not the most annoying thing in the world but I would love to figure this one out. lol

**Comment by blazem** [comment]
Have you tried a rest sensor:


https://github.com/OpenXbox/xboxone-home-assistant/issues/15

**Comment by GsynthChris** [comment]
> Have you tried a rest sensor:
> 
> #15

Well I haven't tried that but now I can't get it to work at all... :(

**Comment by kadiusone** [comment]
Hello, I have same problem in my installation of hass.io

this is what I see in the logs:

Logger: custom_components.xboxone.media_player
Source: custom_components/xboxone/media_player.py:613
Integration: xboxone (documentation)
First occurred: 0:48:44 (3001 occurrences)
Last logged: 10:29:23

Refreshing authentication tokens failed!

Every I have to navigate to ....:5557/auth/oauth to authenticate

Any idea?

**Comment by codehunter13** [comment]
i got the same problem. i run the xbox server inside docker
 Some logs:
127.0.0.1 - - [2020-09-16 18:05:07] "GET /auth/refresh HTTP/1.1" 500 251 0.002451
127.0.0.1 - - [2020-09-16 18:05:07] "GET /auth HTTP/1.1" 200 252 0.001541
[2020-09-16 18:05:07,532] WARNING: Token Auth failed: No valid RefreshToken. Attempting auth via credentials
[2020-09-16 18:05:07,533] ERROR: {'success': False, 'message': 'AuthenticationManager was not able to authenticate with provided tokens or user credentials!'}
127.0.0.1 - - [2020-09-16 18:05:07] "GET /auth/refresh HTTP/1.1" 500 251 0.002485
127.0.0.1 - - [2020-09-16 18:05:14] "GET /versions HTTP/1.1" 200 220 0.069836
127.0.0.1 - - [2020-09-16 18:05:14] "GET /auth HTTP/1.1" 200 252 0.001462
[2020-09-16 18:05:14,979] WARNING: Token Auth failed: No valid RefreshToken. Attempting auth via credentials
[2020-09-16 18:05:14,979] ERROR: {'success': False, 'message': 'AuthenticationManager was not able to authenticate with provided tokens or user credentials!'}
127.0.0.1 - - [2020-09-16 18:05:14] "GET /auth/refresh HTTP/1.1" 500 251 0.002059
127.0.0.1 - - [2020-09-16 18:05:17] "GET /device HTTP/1.1" 200 511 2.527767
127.0.0.1 - - [2020-09-16 18:05:17] "GET /device/FD003B9C19AD9054 HTTP/1.1" 200 489 0.003843
127.0.0.1 - - [2020-09-16 18:05:17] "GET /auth HTTP/1.1" 200 252 0.003140
[2020-09-16 18:05:17,564] WARNING: Token Auth failed: No valid RefreshToken. Attempting auth via credentials
[2020-09-16 18:05:17,565] ERROR: {'success': False, 'message': 'AuthenticationManager was not able to authenticate with provided tokens or user credentials!'}
127.0.0.1 - - [2020-09-16 18:05:17] "GET /auth/refresh HTTP/1.1" 500 251 0.004314
127.0.0.1 - - [2020-09-16 18:05:17] "GET /auth HTTP/1.1" 200 252 0.001953
[2020-09-16 18:05:17,606] WARNING: Token Auth failed: No valid RefreshToken. Attempting auth via credentials
[2020-09-16 18:05:17,607] ERROR: {'success': False, 'message': 'AuthenticationManager was not able to authenticate with provided tokens or user credentials!'}

**Comment by kadiusone** [comment]
Seems I solved it, or at least I do not have to authorize it every day.

When I was asked to authorize (by paste the link) I did that but just after that I logged out and authorized by login and password. Seems to solve every day OAuth.I have to do it some times but still better than every day.

I have also added #15 sensor (mentioned in the thread) but that was not working for me but I have it still active.
